src/blocks/navigation.ts

```diff
@@ -16,12 +22,18 @@ export function fromDTO(dto: JsonObject): PartialNavigation {
 
   return {
     items:
-      dto.items.map((item: NavListItemDTO) => ({
-        id: item.id ?? '',
-        parentId: item.parentId ?? null,
-        title: item.title ?? '',
-        blockId: item.blockId ?? null,
-      })) ?? [],
+      dto.items.map((item: NavListItemDTO) => {
+        if (item.kind === NavListItemKindEnum.Group) {
+          return item;
```

Comment:
Can we copy things out here to ensure the DTO has the right shape/type and nothing else gets through.